File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/opptjening/OpptjeningAssemblerTest.java
package no.nav.pensjon.selvbetjeningopptjening.opptjening;

import no.nav.pensjon.selvbetjeningopptjening.consumer.uttaksgrad.UttaksgradGetter;
import org.junit.jupiter.api.Test;
import org.mockito.Mock;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

class OpptjeningAssemblerTest {

    @Mock
    UttaksgradGetter uttaksgradGetter;

    @Test
    void removeFutureOpptjening() {
        Map<Integer, Opptjening> opptjeningerByYear = new HashMap<>();
        opptjeningerByYear.put(2018, new Opptjening(1L, 1.1D));
        opptjeningerByYear.put(2019, new Opptjening(1L, 1.1D));
        opptjeningerByYear.put(2020, new Opptjening(1L, 1.1D));

        new TestOpptjeningAssembler(uttaksgradGetter).removeFutureOpptjening(opptjeningerByYear, 2019);

        assertEquals(2, opptjeningerByYear.size());
    }

    private static class TestOpptjeningAssembler extends OpptjeningAssembler {

        TestOpptjeningAssembler(UttaksgradGetter uttaksgradGetter) {
            super(uttaksgradGetter);
        }
    }
}

File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/opptjening/OpptjeningAssemblerTest.kt
package no.nav.pensjon.selvbetjeningopptjening.opptjening

import io.kotest.core.spec.style.ShouldSpec
import io.kotest.matchers.maps.shouldHaveSize
import io.mockk.mockk
import no.nav.pensjon.selvbetjeningopptjening.consumer.uttaksgrad.UttaksgradGetter

class OpptjeningAssemblerTest : ShouldSpec({

    should("remove entry for given year") {
        val opptjeningerByYear: Map<Int, Opptjening> = mapOf(
            2018 to opptjening(),
            2019 to opptjening(),
            2020 to opptjening()
        )

        TestOpptjeningAssembler(
            getter = mockk<UttaksgradGetter>()
        ).removeFutureOpptjening(opptjeningerByYear, 2019)

        opptjeningerByYear shouldHaveSize 2
    }
})

private class TestOpptjeningAssembler(getter: UttaksgradGetter) :
    OpptjeningAssembler(getter)

private fun opptjening() =
    Opptjening(1L, 1.1)